android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/PhysicalWebBroadcastService.java

```diff
@@ -0,0 +1,216 @@
+    private BluetoothManager mBluetoothManager;
+    private BluetoothLeAdvertiser mBluetoothLeAdvertiser;
+    private static final int NOTIFICATION_ID = 6;
+    public static final String BROADCAST_URL_KEY = "URL";
+    public static final String DISPLAY_URL_KEY = "OURL";
+    private NotificationManagerCompat mNotificationManager;
+    private IBinder mLocalBinder;
+    private Handler mHandler = new Handler();
+    private String mdisplayUrl;
+    private String shareUrl;
+
+    private final BroadcastReceiver mReceiver = new BroadcastReceiver() {
+        @Override
+        public void onReceive(Context context, Intent intent) {
+            Log.d(TAG, "in receiver");
+            String action = intent.getAction();
+
+            if (BluetoothAdapter.ACTION_STATE_CHANGED.equals(action)) {
+                int state = intent.getIntExtra(BluetoothAdapter.EXTRA_STATE, -1);
+                switch (state) {
+                    case BluetoothAdapter.STATE_OFF:
+                        Log.d(TAG, "stop becuase BT off");
+                        stopSelf();
+                        break;
+                    default:
+
+                }
+            }
+        }
+    };
+
+    /////////////////////////////////
+    // callbacks
+    /////////////////////////////////
+
+    @Override
+    public void onCreate() {
+        super.onCreate();
+        Log.d(TAG, "SERVICE onCreate");
+        mBluetoothManager = (BluetoothManager) getApplicationContext().getSystemService(
+            Context.BLUETOOTH_SERVICE);
+        mBluetoothLeAdvertiser = mBluetoothManager.getAdapter().getBluetoothLeAdvertiser();
+        mNotificationManager = NotificationManagerCompat.from(this);
+    }
+
+    @Override
+    public int onStartCommand(Intent intent, int flags, int startId) {
+        Log.d(TAG, "SERVICE onStartCommand");
+        IntentFilter filter = new IntentFilter(BluetoothAdapter.ACTION_STATE_CHANGED);
+        registerReceiver(mReceiver, filter);
+        // Only call initialize once
+        // We do this because MainActivity launches the service via startService
+        // but it may call it more than once during initialization
+        shareUrl = intent.getStringExtra(BROADCAST_URL_KEY);
+        mdisplayUrl = intent.getStringExtra(DISPLAY_URL_KEY);
+        Log.d(TAG, shareUrl);
+        broadcastUrl();
+        Log.d(TAG, "SERVICE running!");
+
+        // Keep the service running
+        return START_STICKY;
+    }
+
+    @Override
+    public void onDestroy() {
+        Log.d(TAG, "SERVICE onDestroy");
+        unregisterReceiver(mReceiver);
+        disableUrlBroadcasting();
+        super.onDestroy();
+    }
+
+    // Fires when user swipes away app from the recent apps list
+    @Override
+    public void onTaskRemoved (Intent rootIntent) {
+        Log.d(TAG, "onTaskRemoved");
+        stopSelf();
+    }
+
+    @Override
+    public IBinder onBind(Intent intent) {
+        return mLocalBinder;
```

Comment:
mLocalBinder is never set, so it will always be null.  Might as well remove the member and return null instead.